spec/requests/posts_controller_spec.rb

```diff
@@ -0,0 +1,20 @@
+RSpec.describe 'Controllers', type: :request do
+  describe 'PostsController' do
+    describe 'GET #index' do
+      it 'returns a successful response' do
+        get posts_path
+        expect(response).to be_successful
+      end
+
+      it 'renders the index template' do
+        get posts_path
+        expect(response).to render_template(:index)
+      end
+
+      it 'includes correct placeholder text in the response body' do
+        get posts_path
+        expect(response.body).to include('Display posts')
+      end
+    end
+  end
+end
```

Comment:
- [ ] Kindly I recommend you to include test case for **show actions** for User controllers as you write the test case for index actions Just like the below sample and kindly also include the above list three scenario

```
describe 'GET /show' do
  ...
end
 
```